secboot/secboot_sb.go

```diff
@@ -162,6 +164,23 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 		return res, fmt.Errorf("internal error: cannot build an auth requestor: %v", err)
 	}
 
+	// Non-nil FDEHookKeyV1 indicates that V1 hook key is used
+	if loadedKey.FDEHookKeyV1 != nil {
+		// Special case for hook keys v1. They do not have
+		// primary keys. So we cannot wrap them in KeyData
+		err := unlockDiskWithHookV1Key(mapperName, sourceDevice, loadedKey.FDEHookKeyV1)
+		if err == nil {
+			res.FsDevice = targetDevice
+			res.UnlockMethod = UnlockedWithSealedKey
+			return res, nil
+		}
+		// If we did not manage we should still try unlocking
+		// with key data if there are some on the tokens.
+		// Also the request for recovery key will happen in
+		// ActivateVolumeWithKeyData
+		logger.Noticef("WARNING: attempting opening device %s  with key file %s failed: %v", sourceDevice, sealedEncryptionKeyFile, err)
```

Comment:
this fallback path is not reached

Reply:
I have added `TestUnlockVolumeUsingSealedKeyIfEncryptedFdeRevealKeyV1FallbackKeyData